src/BSON/RegexInterface.stub.php

```diff
@@ -0,0 +1,18 @@
+<?php
+
+/** @generate-function-entries */
+
+namespace MongoDB\BSON
+{
+    interface RegexInterface
+    {
+        /** @tentative-return-type */
+        public function getPattern(): string;
+
+        /** @tentative-return-type */
+        public function getFlags(): string;
+
+        /** @tentative-return-type */
+        public function __toString(): string;
```

Comment:
Just to confirm: we need to use tentative return types only for the standalone _interfaces_ but not the concrete classes thanks to PHP's internal handling for Stringable implementations. This relates to https://github.com/mongodb/mongo-php-driver/commit/6c82ec8b52fad33c3e9444ae8bd2c27a729c4688.

Reply:
That is correct. Strictly speaking, the interface should already extend the `Stringable` interface, which would make this a hard BC break. Adding the `string` return type to `__toString` in the implementation of BSON type interfaces removes this BC breaks, as those classes are all final. We can clean all this up in a 2.0 release where we're free to break BC. Until then, the tentative return type definition will alert implementers of the BSON interfaces that we're adding a return type that they should be adding already.